(feat): make supportsInterface function on OptimismMintableERC20 virtual

KNWR opened the pull request:
<!-- Contributions welcome! See https://github.com/ethereum-optimism/.github/blob/master/CONTRIBUTING.md -->

**Description**

Let's say I want to extend OptimismMintableERC20 to support additional functionality with corresponding interfaces. Currently if I want to extend the contract and have the `supportsInterface` support additional interfaces I'm adding, I can't do that. This change makes it possible by marking`supportsInterface` as virtual, so that other contracts can inherit from OptimismMintableERC20 and register as supporting additional interfaces by including a `supportsInterface` method marked as `override`. 

**Tests**

No additional tests needed as this doesn’t affect runtime functionality. Tests still passing.

tynes commented:
Looks like yarn is missing during the build?

```
.._130_489ae780f8fce928019ae73fb8d071ee yarn-install$ yarn install
..._130_489ae780f8fce928019ae73fb8d071ee yarn-install: sh: 1: yarn: not found
 ERR_PNPM_PREPARE_PACKAGE  Failed to prepare git-hosted package fetched from "https://codeload.github.com/Saw-mon-and-Natalie/clones-with-immutable-args/tar.gz/105efee1b9127ed7f6fedf139e1fc796ce8791f2": clones-with-immutable-args@2.0.0 yarn-install: `yarn install`
spawn ENOENT
```

cc @roninjin10 any idea what is going on here?

tynes commented:
I don't believe that this changes the bytecode, if it does not change the bytecode then you can make this a patch semver update. If it does change the bytecode, then you also need to update the minor semver version of the erc20 factory contract

tynes commented:
> Looks like yarn is missing during the build?
> 
> ```
> .._130_489ae780f8fce928019ae73fb8d071ee yarn-install$ yarn install
> ..._130_489ae780f8fce928019ae73fb8d071ee yarn-install: sh: 1: yarn: not found
>  ERR_PNPM_PREPARE_PACKAGE  Failed to prepare git-hosted package fetched from "https://codeload.github.com/Saw-mon-and-Natalie/clones-with-immutable-args/tar.gz/105efee1b9127ed7f6fedf139e1fc796ce8791f2": clones-with-immutable-args@2.0.0 yarn-install: `yarn install`
> spawn ENOENT
> ```
> 
> cc @roninjin10 any idea what is going on here?

Already on top of it, thanks @roninjin10 

https://github.com/ethereum-optimism/optimism/pull/6287

tynes commented:
We need to trigger a new release of `ci-builder`, will come back to this PR

roninjin10 commented:
@tynes you just need to rebase you don't need to wait on ci-builder

KNWR commented:
thanks folks --

@roninjin10 as in I should rebase this branch off of #6303 ("chore: remove unnecessary yarn and pnpm in circle config") or off of the `develop` branch?

tynes commented:
> thanks folks --
> 
> @roninjin10 as in I should rebase this branch off of #6303 ("chore: remove unnecessary yarn and pnpm in circle config") or off of the `develop` branch?

Rebase on develop please

KNWR committed:
feat: Make supportsInterface function on OptimismMintableERC20.sol virtual

This change makes it possible for other contracts that inherit from OptimismMintableERC20 to register as supporting additional interfaces beyond those declared in the contract as it is.

tynes commented:
@KNWR looks like the bindings need to be rebuilt
